opm/models/discretization/common/fvbasefdlocallinearizer.hh

```diff
@@ -247,9 +239,9 @@ public:
         localResidual_.eval(residual_, elemCtx);
 
         // calculate the local jacobian matrix
-        size_t numPrimaryDof = elemCtx.numPrimaryDof(/*timeIdx=*/0);
-        for (unsigned dofIdx = 0; dofIdx < numPrimaryDof; dofIdx++) {
-            for (unsigned pvIdx = 0; pvIdx < numEq; pvIdx++) {
+        const std::size_t numPrimaryDof = elemCtx.numPrimaryDof(/*timeIdx=*/0);
+        for (unsigned dofIdx = 0; dofIdx < numPrimaryDof; ++dofIdx) {
+            for (unsigned pvIdx = 0; pvIdx < numEq; ++pvIdx) {
```

Comment:
If you're changing these lines anyway, maybe we could write it as
```C++
for (auto dofIdx = 0*numPrimaryDof; ...) {
    for (auto pvIdx = 0*numEq; ...) {
```
instead?  That way, we don't incur type conversion on every comparison with the upper bounds.

Reply:
problem is you are just moving the type conversions into the function calls, where both arguments are unsigned. for the first it would do ```std::size_t -> unsigned```, for the second it would do ```int -> unsigned```. So it makes no difference basically. Anyways, I have applied the requested change.